flink-clients/src/main/java/org/apache/flink/client/program/StreamContextEnvironment.java

```diff
@@ -244,12 +240,21 @@ public static void unsetAsContext() {
     }
 
     private List<String> collectNotAllowedConfigurations() {
-        final List<String> errors = new ArrayList<>();
-        if (allowConfigurations) {
-            return errors;
+        if (programConfigEnabled) {
+            return Collections.emptyList();
         }
+
+        final List<String> errors = new ArrayList<>();
+
+        final Configuration clusterConfigMap = new Configuration(originalClusterConfig);
+        final Configuration envConfigMap = new Configuration(configuration);
+
+        removeProgramConfigWildcards(clusterConfigMap);
+        removeProgramConfigWildcards(envConfigMap);
```

Comment:
the whole `removal` / `enrichment` seems overcomplicated;

can we simply check for wildcard keys in the "diff functions" below?